ql/src/test/queries/clientpositive/cast_null_to_complex.q

```diff
@@ -0,0 +1,13 @@
+SET hive.cli.print.header=true;
+
+explain cbo
+select cast(null as map<int, string>), cast(null as array<map<int, string>>), cast(null as int), cast(null as struct<f1:array<map<int, string>>, f2:struct<a:double, b:string>>);
+explain
+select cast(null as map<int, string>), cast(null as array<map<int, string>>), cast(null as int), cast(null as struct<f1:array<map<int, string>>, f2:struct<a:double, b:string>>);
+select cast(null as map<int, string>), cast(null as array<map<int, string>>), cast(null as int), cast(null as struct<f1:array<map<int, string>>, f2:struct<a:double, b:string>>);
+
+
+create table t1 as
+select cast(null as map<int, string>), cast(null as array<map<int, string>>), cast(null as int), cast(null as struct<f1:array<map<int, string>>, f2:struct<a:double, b:string>>);
```

Comment:
Is this select equivalent with a table having the same fields? I mean in the select the cast(null as  <type>) is already there. There should be one more test where a table is created with these complex fields, and the table is referred in the CTAS. That would test if the CAST expressions are generated correctly in the background.
Also, the original issue reported with a sub-query, so that case also should be covered.

Reply:
added q test for ctas from empty source table and a select with a subquery producing empty result